pkg/controller/metadata/metadata_controller.go

```diff
@@ -347,6 +349,44 @@ func (r *ReconcileMetadata) generateMetadataSecretData(instance *verifyv1beta1.M
 	return data, nil
 }
 
+// This function determins if we should regenerate the metadata or not.
+func ShouldRegenerate(secretsObj *corev1.Secret, hashOfSecrets string) bool {
+	secretsMap := secretsObj.Data
+
+	// If the rest of the config has changed and the hash has changed, regenerate regardless.
+	if secretsObj.ObjectMeta.Annotations[VersionAnnotation] != hashOfSecrets {
+		return true
+	}
+
+	// Checking the date from the secrets store to see if we need to regenerate the metadata.
+	byteValidityDays := secretsMap[ValidityDays]
+	byteValidUntil := secretsMap[ValidUntil]
+
+	// If theres nothing in the map the regenerate metadata.
+	if byteValidityDays == nil || byteValidUntil == nil {
+		return true
+	}
+
+	intValidityDays, errValidityDays := strconv.Atoi(string(byteValidityDays))
+	validUntilTimeStamp, errValidUntil := time.Parse(time.RFC1123Z, string(byteValidUntil))
+
+	if errValidUntil != nil || errValidityDays != nil {
+		return true
+	}
+
+	// We want to regenerate the metadata if its atleast halfway through its lifetime
+	metadataLifetime := intValidityDays / 2
+
+	regeneratePastThisDate := time.Now().AddDate(0, 0, metadataLifetime)
+
+	// If the timestamp is after the metadatalife time then regenerate it.
+	if validUntilTimeStamp.After(regeneratePastThisDate) {
```

Comment:
These last few lines could just be `return validUntilTimeStamp.After(regeneratePastThisDate)` if that's allowable in Go.